src/System.Net.Sockets/tests/FunctionalTests/NetworkStreamTest.cs

```diff
@@ -11,6 +11,452 @@ namespace System.Net.Sockets.Tests
+            {
+                if (close) server.Close();
+                else server.Dispose();
+
+                Assert.Throws<ObjectDisposedException>(() => server.DataAvailable);
+
+                Assert.Throws<ObjectDisposedException>(() => server.Read(new byte[1], 0, 1));
+                Assert.Throws<ObjectDisposedException>(() => server.Write(new byte[1], 0, 1));
+
+                Assert.Throws<ObjectDisposedException>(() => server.BeginRead(new byte[1], 0, 1, null, null));
+                Assert.Throws<ObjectDisposedException>(() => server.BeginWrite(new byte[1], 0, 1, null, null));
+
+                Assert.Throws<ObjectDisposedException>(() => server.EndRead(null));
+                Assert.Throws<ObjectDisposedException>(() => server.EndWrite(null));
+
+                Assert.Throws<ObjectDisposedException>(() => { server.ReadAsync(new byte[1], 0, 1); });
+                Assert.Throws<ObjectDisposedException>(() => { server.WriteAsync(new byte[1], 0, 1); });
+
+                Assert.Throws<ObjectDisposedException>(() => { server.CopyToAsync(new MemoryStream()); });
+
+                return Task.CompletedTask;
+            });
+        }
+
+        [Fact]
+        public async Task DisposeSocketDirectly_ReadWriteThrowIOException()
+        {
+            using (Socket listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            using (Socket client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                listener.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+                listener.Listen(1);
+
+                Task<Socket> acceptTask = listener.AcceptAsync();
+                await Task.WhenAll(acceptTask, client.ConnectAsync(new IPEndPoint(IPAddress.Loopback, ((IPEndPoint)listener.LocalEndPoint).Port)));
+                using (Socket serverSocket = await acceptTask)
+                using (DerivedNetworkStream server = new DerivedNetworkStream(serverSocket))
+                {
+                    serverSocket.Dispose();
+
+                    Assert.Throws<IOException>(() => server.Read(new byte[1], 0, 1));
+                    Assert.Throws<IOException>(() => server.Write(new byte[1], 0, 1));
+
+                    Assert.Throws<IOException>(() => server.BeginRead(new byte[1], 0, 1, null, null));
+                    Assert.Throws<IOException>(() => server.BeginWrite(new byte[1], 0, 1, null, null));
+
+                    Assert.Throws<IOException>(() => { server.ReadAsync(new byte[1], 0, 1); });
+                    Assert.Throws<IOException>(() => { server.WriteAsync(new byte[1], 0, 1); });
+                }
+            }
+        }
+
+        [Fact]
+        public async Task InvalidIAsyncResult_EndReadWriteThrows()
+        {
+            await RunWithConnectedNetworkStreamsAsync((server, _) =>
+            {
+                Assert.Throws<IOException>(() => server.EndRead(Task.CompletedTask));
```

Comment:
There is here:
https://github.com/dotnet/corefx/pull/16607/files/6921f1843a2e81f31b3d695a9eed0e9588a811db..98de35603ab4a9a8b28ce952230b7be2392dbefb#diff-8e9d85bace3bf44d6ed8504ba46c1eaaR378